### PR TITLE
Add redundancy pos chg

### PR DESCRIPTION
**Pull request by samstratton13**
- Added a get market position function 
- look back over time window for the latest signal
- React to the latest signal if not already done. Check with the qty of the coin

### CODE REVIEW

File: src/clients/candlesticks.py [hunk 1]
```diff
@@ -117,7 +117,24 @@ def create_ma_crossover_dataframe(self, window_min, window_max, units):
         return main_df
 
     def get_current_position(self, dataframe):
-        return dataframe['Position'].iloc[-1]
+        # return the latest non 0 within values within a time frame
+        match self.candleTimeframe:
+            case "1m":
+                look_back_window = 10 #10 minute look back window for signals, Check the last signal.
```
Comment (TobyCourtis) [inline src/clients/candlesticks.py]: do a quick reformat

File: src/test/test_candlesticks.py [hunk 2]
```diff
@@ -66,28 +66,94 @@ def test_shorten_fails_if_non_int_passed(self):
 
         self.assertRaises(TypeError, candles.shorten, required_length)
 
-    def test_diff_up(self):
+    def test_position_1m_(self):
         df = pd.DataFrame([[0], [1]], columns=['Signal'])
```
Comment (TobyCourtis) [inline src/test/test_candlesticks.py]: Change this to position - technically my fault

File: src/clients/live_ma_crossover_notifier.py [hunk 3]
```diff
@@ -50,21 +50,30 @@ def notify_ma_crossover(window_min, window_max, units):
         print("\n")
 
         position = all_candles.get_current_position(ma_crossover_dataframe)
+        qty = client.get_market_position()
 
-        if position == 1:
+        if (position == 1) & (qty == 0):
```
Comment (TobyCourtis) [inline src/clients/live_ma_crossover_notifier.py:55]: add small comment for what this comparison means

**Review by TobyCourtis — APPROVED** [review]
Small changes so approved - just fix the merge

File: src/test/test_candlesticks.py (hunk 2 above)
Comment (samstratton13) [reply src/test/test_candlesticks.py]: should be good

**Pull request merged**